src/main/java/com/iota/iri/service/tipselection/impl/ConnectedComponentsStartingTipSelector.java

```diff
@@ -1,27 +1,41 @@
 package com.iota.iri.service.tipselection.impl;
 
 import com.iota.iri.controllers.ApproveeViewModel;
+import com.iota.iri.controllers.TipsViewModel;
 import com.iota.iri.controllers.TransactionViewModel;
 import com.iota.iri.model.Hash;
+import com.iota.iri.service.tipselection.StartingTipSelector;
 import com.iota.iri.storage.Tangle;
 
 import java.security.SecureRandom;
 import java.util.*;
 import java.util.stream.Collectors;
 
-public class ConnectedComponentsCalculator {
+public class ConnectedComponentsStartingTipSelector implements StartingTipSelector {
     public final Tangle tangle;
 
     private final int maxTransactions;
+    private final int maxInitialTips;
     private final Random random;
+    private final TipsViewModel tipsViewModel;
 
-    public ConnectedComponentsCalculator(Tangle tangle, int maxTransactions) {
-        this.maxTransactions = maxTransactions;
+    public ConnectedComponentsStartingTipSelector(Tangle tangle, int maxTransactions, TipsViewModel tipsViewModel) {
         this.tangle = tangle;
+        this.maxTransactions = maxTransactions;
```

Comment:
this sounds like a small number - currently 400, especially given that `maxInitialTips` will be 100, so the CC might be very small.

would be good to consult @clara on this, but my personal bias is sqrt(N)

Reply:
Are you suggesting to change
```
this.maxInitialTips = maxTransactions / 4;
```
to
```
this.maxInitialTips = sqrt(maxTransactions);
```
?

Reply:
Yes, well converted to int :)